### PR TITLE
feat: studies list page update

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Show status for studies
* Hide readwrite permission for readonly study
* Disable selection and permission update for pending and unavailable studies 

Existing internal study and ReadOnly external study in available status: 
Screen Shot 2020-12-17 at 3 02 13 PM

External ReadWirte study in pending status:
Screen Shot 2020-12-17 at 3 01 46 PM

External ReadWrite study in error status:
Screen Shot 2020-12-17 at 3 01 32 PM

Checklist: 

* [x] Have you successfully deployed to an AWS account with your changes? 
* [x] Have you linted your code locally prior to submission?
* [x] Have you successfully ran unit tests and manual tests with your changes locally?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by SanketD92 — CHANGES_REQUESTED** [review]
The base branch needs updating on your topic branch.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js [hunk 1]
```diff
@@ -103,7 +107,8 @@ class StudyPermissionsTable extends React.Component {
 
   renderTable() {
     const studyPermissions = this.permissionsStore.studyPermissions;
-    const isEditable = studyPermissions.adminUsers.some(uid => uid === this.currUser.uid);
+    const isEditable =
+      studyPermissions.adminUsers.some(uid => uid === this.currUser.uid) && this.study.state.canChangePermission;
```
Comment (hatimkhan) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js:111]: Is it possible that you move this existing logic as a view on the permissions entity model?

something like  isAdmin(uid)  as a view on the permissions entity model?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/studies/Study.js [hunk 2]
```diff
@@ -56,7 +92,7 @@ const Study = types
 
     getPermissionsStore() {
       if (!self.permissionsStore) {
-        self.permissionsStore = StudyPermissionsStore.create({ studyId: self.id });
+        self.permissionsStore = StudyPermissionsStore.create({ studyId: self.id, studyAccessType: self.accessType });
```
Comment (hatimkhan) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/models/studies/Study.js:95]: Is it possible not to introduce the property 'studyAccessType' at the permissions entity level?  can you make it a derived property, instead?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js (hunk 1 above)
Comment (Bingjiling) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js:111]: Emmm, view methods can not have pass in parameter

Comment (Bingjiling) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js:111]: Ah, my bad. I thought view method can only be `get` something, which can't have pass in parameter. Will update it.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/studies/Study.js [hunk 3]
```diff
@@ -15,10 +15,44 @@
 
 import { types, applySnapshot } from 'mobx-state-tree';
 
+import _ from 'lodash';
 import { StudyFilesStore } from './StudyFilesStore';
 import { StudyPermissionsStore } from './StudyPermissionsStore';
 import { categories } from './categories';
 
+// 'pending', 'error', 'default'
+// Note: When study is available, the status column is empty
```
Comment (hatimkhan) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/models/studies/Study.js:24]: Is that true?  That is only in the database, but when we convert the dbEntity to a studyEntity in the backend we populate the status as 'reachable' if there is no status in the database.  Can you confirm? thanks

**Review by hatimkhan — APPROVED** [review]
return _.map(result, toStudyEntity);

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**